pkg/log/log.go

```diff
@@ -117,7 +120,24 @@ func (l *logger) Print(msg string, fields ...field.M) {
 
 	entry := log.WithFields(logFields)
 	if l.err != nil {
-		entry = entry.WithError(l.err)
+		switch e := l.err.(type) {
+		case awserr.Error:
+			errFields := make(logrus.Fields)
+			errFields["awsErrorCode"] = e.Code()
+			errFields["awsErrorMessage"] = e.Message()
+			if er, ok := e.(awserr.RequestFailure); ok {
+				errFields["awsRequestStatusCode"] = er.StatusCode()
+				errFields["awsRequestID"] = er.RequestID()
+			}
+			if nextErr := e.OrigErr(); nextErr != nil {
```

Comment:
What is this doing with OrigErr and the "error"  Field?

Reply:
`OrigErr` returns the original error if one was set.  Nil is returned if not set.
Since we are not calling entry.WithError() on the awserr which sets the "error" field to the error passed as parameter, we are setting it. this will assign the field `error` with value `nextErr` in the error log.

entry.WithError(err) internally calls entry.WithField(ErrorKey, err). in the case of awserrs we are extracting the fields and setting them.